spec/Section 3 -- Type System.md

```diff
@@ -1319,8 +1336,8 @@ Union types have the potential to be invalid if incorrectly defined.
 
 1. A Union type must include one or more unique member types.
 2. The member types of a Union type must all be Object base types;
```

Comment:
Given this and other conversations around "concrete" types (e.g. re `__typename`) makes me wonder if we should RFC the idea of a "concrete" type into the spec everywhere as a prerequisite before we add a tagged type?

Reply:
Initially I was hesitant to add Tagged to Unions; but I like this idea @leebyron, particularly because it solves the "union of objects and scalars" problem in a relatively clean way:

```graphql
type City {
  id: ID!
  name: String!
  coordinates: Coordinates!
}

type Town {
  id: ID!
  name: String!
  coordinates: Coordinates!
}

type POI {
  id: ID!
  name: String!
  coordinates: Coordinates!
  nearestCity: City!
}

type Coordinates {
  latitude: Float!
  longitude: Float!
}

tagged LocationSpecifier {
  locationName: String!
  locationCoordinates: Coordinates!
}

union Place = City | Town | POI | LocationSpecifier;

type Query {
  myFavoritePlace: Place
}
```

Now your favourite place could be a known city, town, point of interest, a custom "named place", or a custom position on the globe.

Let's leave it for a follow up PR; I don't _think_ that it would need any breaking changes... Do you?